source/fuzz/fuzzer.cpp

```diff
@@ -383,23 +332,35 @@ bool Fuzzer::ShouldContinueFuzzing() {
     // Stop because we have reached the transformation limit.
     return false;
   }
-  // If we have applied T transformations so far, and the limit on the number of
-  // transformations to apply is L (where T < L), the chance that we will
-  // continue fuzzing is:
-  //
-  //     1 - T/(2*L)
-  //
-  // That is, the chance of continuing decreases as more transformations are
-  // applied.  Using 2*L instead of L increases the number of transformations
-  // that are applied on average.
-  auto chance_of_continuing = static_cast<uint32_t>(
-      100.0 * (1.0 - (static_cast<double>(transformations_applied_so_far) /
-                      (2.0 * static_cast<double>(
-                                 fuzzer_context_->GetTransformationLimit())))));
-  if (!fuzzer_context_->ChoosePercentage(chance_of_continuing)) {
-    // We have probabilistically decided to stop.
+  assert(transformations_applied_so_far >= initial_num_of_transformations &&
```

Comment:
Above, on lines 323-327, is the code where we are once again checking whether the fuzzer got stuck. I'm proposing removing this, and by moving the stuck check of "Run" into the do ... while loop we will check it there and return the special stuck status.

Reply:
Fixed in 5967f1f.